### PR TITLE
feat: add batches API with OpenAI compatibility

### PR DESCRIPTION
**Pull request by mattf**
Add complete batches API implementation with protocol, providers, and tests:

Core Infrastructure:
- Add batches API protocol using OpenAI Batch types directly
- Add Api.batches enum value and protocol mapping in resolver
- Add OpenAI "batch" file purpose support
- Include proper error handling (ConflictError, ResourceNotFoundError)

Reference Provider:
- Add ReferenceBatchesImpl with full CRUD operations (create, retrieve, cancel, list)
- Implement background batch processing with configurable concurrency
- Add SQLite KVStore backend for persistence
- Support /v1/chat/completions endpoint with request validation

Comprehensive Test Suite:
- Add unit tests for provider implementation with validation
- Add integration tests for end-to-end batch processing workflows
- Add error handling tests for validation, malformed inputs, and edge cases

Configuration:
- Add max_concurrent_batches and max_concurrent_requests_per_batch options
- Add provider documentation with sample configurations

Test with -

```
$ uv run llama stack build --image-type venv --providers inference=YOU_PICK,files=inline::localfs,batches=inline::reference --run &
$ LLAMA_STACK_CONFIG=http://localhost:8321 uv run pytest tests/unit/providers/batches tests/integration/batches --text-model YOU_PICK
```

addresses #3066

### CODE REVIEW

**Comment by mattf** [comment]
@franciscojavierarceo ptal

**Comment by ashwinb** [comment]
lgtm

I am not a fan of the "batches" name for an API but hey this time I can just ignore that piece and point to someone else :)

**Comment by mattf** [comment]
> Should we note in the documentation somewhere that this API is under development?

i updated provider_codegen.py to include the api protocol's docstring in the providers docs and added a note about the development status.

fyi, this also picked up additional docs from agents, inference & eval.

**Comment by mattf** [comment]
@ashwinb thoughts on the replay tests failing?

File: llama_stack/apis/batches/batches.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    Note: This API is currently under active development and may undergo changes.
+    """
+
+    @webmethod(route="/openai/v1/batches", method="POST")
+    async def create_batch(
+        self,
+        input_file_id: str,
+        endpoint: str,
+        completion_window: str,
+        metadata: dict[str, str] | None = None,
```
Comment (franciscojavierarceo) [inline llama_stack/apis/batches/batches.py:47]:
```suggestion
        metadata: dict[str, Any] | None = None,
```

File: llama_stack/apis/batches/batches.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+    Note: This API is currently under active development and may undergo changes.
+    """
+
+    @webmethod(route="/openai/v1/batches", method="POST")
+    async def create_batch(
+        self,
+        input_file_id: str,
+        endpoint: str,
+        completion_window: str,
```
Comment (franciscojavierarceo) [inline llama_stack/apis/batches/batches.py]:
```suggestion
        completion_window: Literal["24h"],
```

File: llama_stack/providers/inline/batches/reference/batches.py [hunk 3]
```diff
@@ -0,0 +1,553 @@
...
+
+        if completion_window != "24h":
+            raise ValueError(
+                f"Invalid completion_window: {completion_window}. Supported values are: 24h. Code: invalid_value. Param: completion_window",
+            )
+
+        batch_id = f"batch_{uuid.uuid4().hex[:16]}"
+        current_time = int(time.time())
+
+        batch = BatchObject(
```
Comment (franciscojavierarceo) [inline llama_stack/providers/inline/batches/reference/batches.py:150]: It could be useful to generate an idempotency key for retries and avoiding processing the same file more than once.

File: llama_stack/apis/batches/batches.py (hunk 1 above)
Comment (mattf) [reply llama_stack/apis/batches/batches.py:47]: for compat, this definitely needs to be str

File: llama_stack/apis/batches/batches.py (hunk 2 above)
Comment (mattf) [reply llama_stack/apis/batches/batches.py]: thanks, i missed this

File: llama_stack/providers/inline/batches/reference/batches.py (hunk 3 above)
Comment (mattf) [reply llama_stack/providers/inline/batches/reference/batches.py:150]: :heart: idempotency is hugely valuable for apis like this.

since files are immutable, the params can serve as that id.

let me add an issue for that feature.

**Review by franciscojavierarceo — APPROVED** [review]
🚀

File: llama_stack/providers/inline/batches/reference/batches.py (hunk 3 above)
Comment (franciscojavierarceo) [reply llama_stack/providers/inline/batches/reference/batches.py:150]: >since files are immutable, the params can serve as that id.

💯

Comment (mattf) [reply llama_stack/providers/inline/batches/reference/batches.py:150]: https://github.com/llamastack/llama-stack/issues/3144
